src/main/java/md/maib/service/impl/CustomerServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package md.maib.service.impl;
+
+import md.maib.entity.Customer;
+import md.maib.repository.CustomerRepository;
+import md.maib.service.CustomerService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class CustomerServiceImpl implements CustomerService {
```

Comment:
As I remember we discussed to migrate to clear DAO without Spring Data. It will be more better for beginners. Please take a look